src/libsyncengine/update_detection/file_system_observer/snapshot/snapshot.cpp

```diff
@@ -187,7 +187,18 @@ NodeId Snapshot::itemId(const SyncPath &path) const {
 
         bool idFound = false;
         for (const NodeId &childId: itemIt->second.childrenIds()) {
-            if (name(childId) == *pathIt) {
+            SyncName childName;
+            if (normalizedSearch) {
+                if (!Utility::normalizedSyncName(name(childId), childName)) {
+                    LOGW_WARN(Log::instance()->getLogger(),
+                              L"Error in Utility::normalizedSyncName: " << Utility::formatSyncName(name(childId)));
+                    continue;
+                }
+            } else {
+                childName = name(childId);
+            }
+
+            if (childName == *pathIt) {
```

Comment:
As we store the normalize name in snapshotItem, we could probably do something like this:

```suggestion
            if ((normalizedSearch ? normalizedName(childId) : name(childId)) == *pathIt) {
```
And add the `normalizedName(itemId&)` method to Snapshot class wich will basically be a copy of `name(ItemId&)` but calling `normalizedName()` on the `SnapshotItem`

Reply:
New solution implemented => out of date